Data-Structures/Linked-List/CycleDetectionII.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * A LinkedList based solution for finding the starting node of the cycle in a list.
+ * @returns the node where cycle begins in the linked list. If there is no cycle present, returns null.
+ * @see https://en.wikipedia.org/wiki/Cycle_detection
+ * @see https://leetcode.com/problems/linked-list-cycle-ii/
+ */
+
+function detectCycleNode(head) {
+  let length = 0
+  let fast = head
+  let slow = head
+
+  while (fast !== null && fast.next !== null) {
+    fast = fast.next.next
+    slow = slow.next
+    if (fast === slow) {
+      // Calculate length of the cycle
+      length = lengthCycle(slow)
+      break
+    }
+  }
+
+  if (length === 0) {
+    // If there is no cycle, return null.
+    return null
+  }
+
+  fast = slow = head
```

Comment:
This is what confused me. Please don't reuse these variables for something entirely different...

Given that the pointers move at the same speed, but one is "ahead" of the other (until they meet), maybe call these "ahead" and "behind"? The names "slow" and "fast" are misleading.